chaoslib/litmus/cpu_hog/pod_cpu_hog.yml

```diff
@@ -0,0 +1,144 @@
+            executable: /bin/bash
+          register: container
+  
+        - name: Record the app_container
+          set_fact:
+            c_container: "{{ container.stdout }}"
+  
+      when: c_container is not defined or c_container == ''
+
+    - name: Obtain Container ID of the target container
+      shell: >
+        kubectl get pods {{ app_pod }} -n {{ app_ns }} 
+        -o jsonpath='{.status.containerStatuses[?(@.name=="{{ c_container }}")].containerID}'
+        | awk -F 'docker://' '{print $2}' 
+      args:
+        executable: /bin/bash
+      register: container_id
+
+    - name: Record the app_container_id
+      set_fact:
+        c_container_id: "{{ container_id.stdout }}"
+
+    - name: Derive chaos duration in seconds (from ms)
+      set_fact:
+        c_duration_sec: "{{ ((c_duration|int)/1000)| int }}"
+    
+    - name: Patch the chaos params to cpu stress job template
+      template:
+        src:  /chaoslib/litmus/cpu_hog/app_cpu_stress.j2
+        dest: /chaoslib/litmus/cpu_hog/app_cpu_stress.yml
+      vars:
+        cpu_stress_image: "{{ lib_image }}"
+
+    - name: Setup cpu chaos infrastructure
+      shell: >
+        kubectl create -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n {{ app_ns }}
+      args: 
+        executable: /bin/bash
+      register: cpu_app_deploy_result
+
+    - name: Wait until the cpu stress job is completed
+      shell: >
+        kubectl get pods -l job-name=app-cpu-stress-{{ run_id }} --no-headers -n {{ app_ns }}
+        --no-headers -o custom-columns=:status.phase
+      args: 
+        executable: /bin/bash
+      register: result
+      until: "result.stdout == 'Succeeded'"
+      delay: 1
+      ## considering duration in milliseconds from spec
+      ## with a grace period of 1 min before cpu stress job terminate
+      retries: "{{ ((c_duration|int)/1000 + 120)| int }}"
+
+    - name: Tear down cpu chaos infrastructure
+      shell: >
+        kubectl delete -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n {{ app_ns }} 
+      args:
+        executable: /bin/bash
+
+    - name: Confirm that the cpu stress job is deleted successfully
+      shell: >
+        kubectl get pods -l job-name=app-cpu-stress-{{ run_id }} --no-headers -n {{ app_ns }}
+      args:
+        executable: /bin/bash
+      register: result
+      until: "'No resources found' in result.stderr"
+      delay: 5
+      retries: 60
+
+  rescue: 
+
+    - block: 
+
+        - name: Tear down cpu chaos infrastructure, if setup
+          shell: >
+            kubectl delete -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n {{ app_ns }} 
+          args:
+            executable: /bin/bash
+          when: cpu_app_deploy_result.rc == 0
+        
+        - name: Confirm that the cpu stress job is not present
+          shell: >
+            kubectl get pods -l job-name=app-cpu-stress-{{ run_id }} --no-headers -n {{ app_ns }}
+          args:
+            executable: /bin/bash
+          register: result
```

Comment:
we can write a different register variable according to the task `result` is being used three times and we are just overwriting it if I'm not wrong.

Reply:
if its scope is within this block - and we are not referring it elsewhere it should be fine?